### DISCUSSION TITLE
일단 공지

### DISCUSSION THREAD

**Discussion by uotalkie** [uotalkie/dol-kr#152] [Announcements]
여기도 열려있는 곳이고, 무료사용은 public을 private으로 돌리지 못하는 것 같으니 주의해 사용해 주세요.

**Comment by cyannn1** [comment]
안녕하세요. 
깃걷쪽의 빌드에는 조치를 따로 취하지 않아도 괜찮을까요?

**Reply by uotalkie** [reply]
gitgud 쪽이라면 apk 빌드 말씀이시죠? 그쪽은 제가 아니라 다른 분께서 관리하는 부분이라 그분께서 조치를 취하시는 것을 기대하는 수 밖에 없습니다. gitgud 도 해외 서비스고 하니 괜찮지 않을까 하는 생각이 있기는 합니다만...

**Comment by NullableSafety** [comment]
Public을 Private로 바꾸지 못하는건 현재 레포지토리가 Fork 레포지토리라서 그렇습니다. 바꾸고싶다면 
1. 현재 레포를 내리고 새로 만드셔야합니다. 다만 그러면 PR 히스토리도 다 날라갑니다.
2. Github Support에 문의해서 받는 방법도 있습니다. 다만 요청하기가 조금 애매한 상황입니다.

혹시 모르니 1번 방법 문서URL 드립니다. https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/detaching-a-fork
다만 전부 진행하지는 마시고 해당 문서에서 2번까지만 일단 진행하세요. 모든 커밋/태그들을 다 백업하는 커맨드입니다. (이슈/PR은 제외입니다.) 혹시 모르니까 히스토리 백업겸 해보시는게 좋을듯 합니다.

**Reply by NullableSafety** [reply]
2번 시도해보고싶으시면https://support.github.com/request?q=unfork 가셔서 Virtual Agent -> Request Detach/Extract -> 대충 영어로 `Extract 원한다. 이미 기존 레포는 개발이 진행되지 않고 있고 이쪽에서 메인으로 진행중이다` 정도로 써주시면 됩니다.

**Reply by uotalkie** [reply]
@NullableSafety 
일단 소스는 다른 번역자 분들의 접근성 문제도 있고 소스파일만 가지고는 게임할 수 있는 방법이 있는 것도 아니라 일정 과정을 거쳐야 할 테니 그냥 퍼블릭으로 둘까 하고 있기는 합니다. 깃헙에서 문제가 된다면 깃랩이나 깃구드? 로 이전할 생각이 있긴 합니다만.. 어쨌든 방법 찾아주셔서 감사합니다.

**Comment by cyannn1** [comment]
안녕하세요, git 사용이 처음이다보니 어려워서 질문 드립니다.
인터넷에서 찾아본 내용들을 보면 다음과 같은 과정으로 pull request 적용을 하는 것 같은데, 혹시 잘못되거나 빠진 과정이 있는지 확인해주실 수 있을까요?

1.(git desktop 기준)올라온 pull request의 번호를 확인하고, git merge pr-xxx 명령어를 git bash에 입력해서 develop 브랜치에 합침
2.develop 브랜치에 올라간 번역 파일들을 검토하면서 오타,오류 등을 수정 (작업 상황을 중간에 commit 할 경우 메모에 파일 경로-몇 번째 줄 까지 검수하였는지 명시)
3.올라온 모든 파일들을 검수하였으면 완료하였다고 명시하고 commit

이 순서대로 진행하는 것이 맞을까요?
항상 감사합니다!

**Reply by uotalkie** [reply]
제 경우 깃헙 데스크탑을 쓰는데 그 기준으로 이렇게 합니다.
1. 웹 인터페이스의 pull request merge 창에서 open this in Github Desktop 을 누릅니다.
![gd1](https://github.com/user-attachments/assets/65e2ca7f-43d1-4de4-8721-c2cebbb40384)
2. 그러면 Recent branches에 이렇게 pr/xxx 로 뜹니다.
![gd2](https://github.com/user-attachments/assets/945b62fb-4a3e-41e4-8b34-9a9aebdcdeb0)
3. develop 브랜치로 변경한 다음 Branch 메뉴의 Merge into current branch... 창에서 pr/xxx 를 선택해 merge 합니다.
![gd3](https://github.com/user-attachments/assets/5cd21514-269a-4902-ac85-e31a16fbe859)
4. conflict 가 있다면 conflict 창이 뜨는데 Visual Studio Code 로 conflict 일어난 곳을 찾아서 적절히 conflict를 해소해 줍니다.
5. merge 할 때는 comment를 쓸 수가 없어서 (혹은 제가 모르는 걸지도..) 첫번째 검수 commit 에 pull request 한 부분과 작업한 분 id 등을 적어둡니다. 어디까지 검수했다거나 검수 완료 등의 comment는 제가 어디까지 검수했는지 잊어버리지 않으려고 추가하는 겁니다만 공동작업하게 되면 제대로 적어둬야 할 지도 모르겠네요.

지금까지는 저 혼자 했으니 대충 해도 됐지만 이제 여러 사람이 작업하게 되면 잘못하면 commit이 꼬일 수 있으니 develop에서 갈라지는 develop-cyannn1 같은 브랜치를 새로 만들어서 작업완료후 develop 에서 묶고 그걸 main으로 다시 merge 하는 방법을 쓰던가 해야겠네요. main 과 develop을 protected 처리해야 할 듯?

**Reply by cyannn1** [reply]
이미지까지 포함하여 상세하게 설명해주셔서 감사합니다!

develop-(아이디) 같은 별도 개인 브랜치를 만들어서 작업하는 것이 좋아 보여요! 말씀하신 것 처럼 commit이 꼬이는 문제도 방지할 수 있고 제가 뭐 잘못 건드려도 수습하기 쉬울테니 안정감이 드네요.
develop-cyannn1 브랜치를 만들어서 진행하도록 하겠습니다~



그리고 github desktop 기준이지만, merge 하고서 바로 undo 버튼을 누르면 commit하기 직전의 작성중 상태로 돌아오는데
![20240721_143757](https://github.com/user-attachments/assets/7859d039-b49e-4cab-a6f0-b916c1cea8dd)


여기서 내용을 입력할 수 있나 보네요.
![20240721_143835](https://github.com/user-attachments/assets/c24eb70d-6ee9-457e-87b0-eb845c7bea7c)
![20240721_144117](https://github.com/user-attachments/assets/a4848caf-f769-48f6-b95d-c0cb1e6db069)

이렇게 해도 문제가 없다면 이 방식으로 해보겠습니다! 감사합니다~

**Comment by cyannn1** [comment]
@uotalkie 이렇게 하는게 맞는지 약간 불안하긴 하지만....
pr/150, pr/151, pr/154 검수 작업 진행하였고, 오늘 5.1x 버전 merge 시 발생하는 conflict까지 안뜨게 진행했습니다!
develop-cyannn1 브랜치에서 작업한 상황인데, 여기서 바로 develop 브랜치로 올려도 될까요? 아니면 총괄님이 체크하신 후 merge하시는 게 좋을까요?

**Reply by uotalkie** [reply]
> @uotalkie 이렇게 하는게 맞는지 약간 불안하긴 하지만.... pr/150, pr/151, pr/154 검수 작업 진행하였고, 오늘 5.1x 버전 merge 시 발생하는 conflict까지 안뜨게 진행했습니다! develop-cyannn1 브랜치에서 작업한 상황인데, 여기서 바로 develop 브랜치로 올려도 될까요? 아니면 총괄님이 체크하신 후 merge하시는 게 좋을까요?

@cyannn1 @mansus-traveler  
pull request 하나 검수가 끝나면 바로 develop 브랜치로 merge 해 주시기 바랍니다. 
그렇지 않으면 어떤 request가 끝나고 어떤 request 가 작업중인지 알 수가 없을 것 같네요.

한번 검수된 것을 다시 보는 것은 시간낭비이니 오류체크는 따로 안 할 겁니다. 그건 나중에 버그제보 나오면 그때 고치는 것으로..
그러니 컴파일 해 보시고 해당 내용이 제대로 작동하는지 확인해 보시거나 테스트 페이지에서 테스트해보시는 것을 권장합니다.
제가 보통 쓰는 것은 base-debug 폴더의 custom-testroom.twee 파일입니다. 제가 이것저것 테스트해본다고 좀 지저분한데
대충 지워놓고 테스트하셔도 됩니다. 테스트 룸에 들어가는 방법은 디버그 모드에서 침실에 있습니다. 
아시는지 모르지만 첫 화면에서 "묘" 를 한번 누르고 게임 시작을 하시면 디버그 모드로 시작되니 디버그용 세이브파일을
하나 만들어 두시고 필요에 따라 로드하시면 편합니다.

그리고 제가 discussions 를 자주 보는 게 아니니 궁금한 것이 있으면 그냥 채널의 개발 탭이나 질문 탭에 적어주시거나
채널에 올리기 뭣하면 여기에 글을 올리시고 배포글에 덧글로 discussions 확인 부탁한다고 해 주시기 바랍니다.

일단은 develop-cyannn1 브랜치를 참고해서 pr/150, pr/151, pr/154 완료, pr/155 작업중이라고 알고 있겠습니다. 
끝나면 develop 에 merge 시켜 주세요.

**Reply by cyannn1** [reply]
넵 알겠습니다!
